pandas/core/internals/array_manager.py

```diff
@@ -0,0 +1,902 @@
+"""
+Experimental manager based on storing a collection of 1D arrays
+"""
+from typing import TYPE_CHECKING, Any, Callable, List, Optional, Tuple, TypeVar, Union
+
+import numpy as np
+
+from pandas._libs import algos as libalgos, lib
+from pandas._typing import ArrayLike, DtypeObj, Label
+from pandas.util._validators import validate_bool_kwarg
+
+from pandas.core.dtypes.cast import find_common_type, infer_dtype_from_scalar
+from pandas.core.dtypes.common import (
+    is_bool_dtype,
+    is_dtype_equal,
+    is_extension_array_dtype,
+    is_numeric_dtype,
+)
+from pandas.core.dtypes.dtypes import ExtensionDtype
+from pandas.core.dtypes.generic import ABCDataFrame, ABCSeries
+from pandas.core.dtypes.missing import isna
+
+import pandas.core.algorithms as algos
+from pandas.core.arrays import ExtensionArray, PandasDtype
+from pandas.core.arrays.sparse import SparseDtype
+from pandas.core.construction import extract_array
+from pandas.core.indexers import maybe_convert_indices
+from pandas.core.indexes.api import Index, ensure_index
+from pandas.core.internals.base import DataManager
+from pandas.core.internals.blocks import make_block
+
+if TYPE_CHECKING:
+    from pandas.core.internals.managers import SingleBlockManager
+
+
+T = TypeVar("T", bound="ArrayManager")
+
+
+class ArrayManager(DataManager):
+    """
+    Core internal data structure to implement DataFrame and Series.
+
+    Alternative to the BlockManager, storing a list of 1D arrays instead of
+    Blocks.
+
+    This is *not* a public API class
+
+    Parameters
+    ----------
+    arrays : Sequence of arrays
+    axes : Sequence of Index
+    do_integrity_check : bool, default True
+
+    """
+
+    __slots__ = [
+        "_axes",
```

Comment:
why is _axes private? just a bit odd

Reply:
i see your comment but i am not sure this is obvious at all and would just call it .axes

Reply:
I will add a comment here as well, to clarify. But I can't change it to `axes`, since that is used as "public internal API" outside of internals in pandas, and it needs to be behave the same for BlockManager and ArrayManager

Reply:
hmm, ok